autogpt_platform/backend/backend/data/block.py

```diff
@@ -492,6 +510,135 @@ def get_blocks() -> dict[str, Type[Block]]:
     return load_all_blocks()
 
 
+def is_block_provider_configured(block_cls: type["Block"]) -> bool:
+    """
+    Check if a block has a valid authentication method configured at runtime.
+
+    For example if a block is an OAuth-only block and there env vars are not set,
+    do not show it in the UI.
+
+    """
+    from backend.sdk.registry import AutoRegistry
+
+    # Create an instance to access input_schema
+    try:
+        block = block_cls.create()
+    except Exception as e:
+        # If we can't create a block instance, assume it's not OAuth-only
+        logger.error(f"Error creating block instance for {block_cls.__name__}: {e}")
+        return True
+    logger.debug(
+        f"Checking if block {block_cls.__name__} has a valid provider configured"
+    )
+
+    # Get all credentials fields from input schema
+    credentials_fields = block.input_schema.get_credentials_fields()
+    if not credentials_fields:
+        logger.debug(
+            f"Block {block_cls.__name__} has no credentials fields - Treating as valid"
+        )
+        return True
+
+    # Check required credentials fields
+    required_fields: dict[str, CredentialsMetaInput] = (
+        block.input_schema.get_required_fields()
+    )
+    required_credentials = [f for f in credentials_fields if f in required_fields]
+
+    if len(required_credentials) == 0:
+        logger.warn(
+            f"Block {block_cls.__name__} has no required credentials fields - therefore will worth without credentials configured"
+        )
```

Comment:
1. not a problem
2. typo
```suggestion
        logger.debug(
            f"Block {block_cls.__name__} has no required credentials fields - therefore will work without credentials configured"
        )
```

although: if a credentials input is not required and also not configured, should it be hidden? do we have a mechanism for this?